### ISSUE TITLE
Add spider visibility options on online lists

### ISSUE THREAD

**Issue by dvz** [mybb/mybb#3935]
- show/hide spiders on the index/portal online list
- display order on index/portal online list (before/after logged in users)
- include/exclude spiders in displayed guest counts (may be dependent on the previous setting)

**Comment by Sama34** [comment]
I think we could do with a global radio setting for `Display first, Display last, Don't display`.

> display order on index/portal online list (before/after logged in users)

Do we even need a setting for this? If so, maybe a group display order setting or even use the default group display order value (good opportunity to unlock default groups order).

**Comment by dvz** [comment]
Sounds like a good direction, but note that intended use for group order now (I believe only `showteam.php`) may differ from preferences for the online list, and may more more complex than simple group order (e.g. mixing groups by ordering by username, but showing the new _Bots_ group last).

Online count aside:
- simple approach: Spiders: _Display first_, _Display last_, _Don't display_
- simpler approach: Spiders: _Display last_, _Don't display_
- extensible approach: add _online list order_ and _Display_ checkbox to user groups (last position for new _Bots_ group by default, admins can uncheck the box to hide it completely; same order would mix groups)

  It would, however, interfere with an assumption that online lists are as accurate [as possible](https://github.com/mybb/mybb/issues/3899) and introduce unnecessary features (we only need to change how a category of guests is handled).

**Comment by Sama34** [comment]
> simple approach: Spiders: Display first, Display last, Don't display

I would go with this to remove the need for two settings (first two points in your OP).

**Comment by dvz** [comment]
I imagine it would be a `<select>` if we go with 3 options, but with #3934 it may make sense to remove _display first_ altogether.

**Comment by Eldenroot** [comment]
Good idea! Personally I do not need to display bots in online list, so I will use this setting :)

**Comment by Eldenroot** [comment]
@dvz - this was fixed by my PR---so this can be closed.

**Comment by arcalys** [comment]
The added setting is a yes/no, not a select, should we keep it this way and close the issue?

**Comment by Sama34** [comment]
Seems like a setting to affect the display order is pending. We could use the same setting @Eldenroot added :

`0` don't show spiders
`1` show normally
`2` show after users

Etc.
